Graphs/KannsAlgorithm.js

```diff
@@ -0,0 +1,44 @@
+/**
+ * @author {RaviSadam}
+ * @name kannsAlgorithm
+ * @description -
+ * Kann's Algorithm implementation in JavaScript
+ * @summary
+ * Kann's Algorithm is used for topological sorting in directed acyclic graphs
+ *
+ * @param graph - Graph [[v1,v2],[v3,v4,v5]..]
+ * @param n - number of vertices
+ * @returns {Array} - Empty array if cycle is detected or else result array;
+ *
+ */
+
+export function kannsAlgorithm(graph, n) {
+  if (n === null || n === undefined) throw Error('Invalid n was given')
+  const inorder = Array(n).fill(0)
+  const result = []
```

Comment:
(By the way, you could preallocate this array.)

Reply:
Preallocate could increase the time complexity. At the end we need to check array filled or not.

Reply:
> Preallocate could increase the time complexity.

No, not really. It's still linear time in the size of the graph - O(n + m) where n is the number of nodes and m is the number of edges. (Allocation can also optimistically be assumed to be constant time.)

> At the end we need to check array filled or not.

Yes, and that can be done in constant time, if you keep an index to the last element. It'd look something like this:

```javascript
const topo_order = Array(n) // preallocate
const idx = 0
...
// to push:
topo_order[idx++] = node
...
if (idx < n) throw new Error("no topological order exists (graph is not acyclic)");
```

Or another variant to write this would be:

```javascript
const topo_order = Array(n) // preallocate
for (let i = 0; i < n; ++i) {
    if (stack.length === 0) throw new Error("no topological order exists (graph is not acyclic)");
    const node = stack.pop();
    topo_order[i] = node;
    ...
}
```

The benefit is that when a topological order exists (which should usually be the case; this is the "happy path"), you can save some reallocations of the array.

But this is just a suggestion, I'm fine with pushing to keep the code simple as well. Though perhaps the preallocation variant is a bit more readable, since it makes explicit that we want to have all n nodes be in this array.